src/unix/core.c

```diff
@@ -927,6 +927,11 @@ int uv__io_active(const uv__io_t* w, unsigned int events) {
 }
 
 
+int uv__fd_exists(uv_loop_t* loop, int fd) {
+  return (unsigned) fd < loop->nwatchers && loop->watchers[fd] != NULL;
```

Comment:
why the `(unsigned)`?

Reply:
Avoids an explicit check for values < 0, the cast wraps them around to a value > INT_MAX (which no fd can be and hence `loop->nwatchers` can't be either.)